From: Olaf Hering <olh@suse.de>
Subject: reduce kernel text size

A biarch compiler with default target 64bit will optimize for POWER4. 
This leads to many nops in the code (beside different instruction scheduling).
These nops are useless on a 32bit cpu. Force the old compiler default
to reduce the text size of the kernel.

   text    data     bss     dec     hex filename
4045745  582424  522948 5151117  4e998d ../O2-default/vmlinux
3816086  582424  522948 4921458  4b1872 ../O2-powerpc/vmlinux

Target: powerpc64-suse-linux
Configured with: ../configure --enable-threads=posix --prefix=/usr --with-local-prefix=/usr/local --infodir=/usr/share/info --mandir=/usr/share/man --libdir=/usr/lib --libexecdir=/usr/lib --enable-languages=c,c++,objc,f95,java,ada --enable-checking --with-gxx-include-dir=/usr/include/c++/4.0.2 --enable-java-awt=gtk --disable-libjava-multilib --with-slibdir=/lib --with-system-zlib --enable-shared --enable-__cxa_atexit --without-system-libunwind --with-cpu=default32 --host=powerpc64-suse-linux
Thread model: posix
gcc version 4.0.2 20050818 (prerelease) (SUSE Linux)


Signed-off-by: Olaf Hering <olh@suse.de>

Index: linux-2.6.13-rc6-git13/arch/ppc/Makefile
===================================================================
--- linux-2.6.13-rc6-git13.orig/arch/ppc/Makefile
+++ linux-2.6.13-rc6-git13/arch/ppc/Makefile
@@ -36,6 +36,7 @@ endif
 cpu-as-$(CONFIG_PPC64BRIDGE)	+= -Wa,-mppc64bridge
 cpu-as-$(CONFIG_4xx)		+= -Wa,-m405
 cpu-as-$(CONFIG_6xx)		+= -Wa,-maltivec
+cpu-as-$(CONFIG_6xx)		+= -mcpu=powerpc
 cpu-as-$(CONFIG_POWER4)		+= -Wa,-maltivec
 cpu-as-$(CONFIG_E500)		+= -Wa,-me500
 cpu-as-$(CONFIG_E200)		+= -Wa,-me200
